core/sqf/src/seabed/src/Makefile

```diff
@@ -55,6 +55,16 @@ LIBSBUTIL	= $(LIBEXPDIR)/libsbutil.so
 LIBSQSTATESB	= $(LIBEXPDIR)/libsqstatesb.so
 # neet -lrt for clock_gettime
 LIBSX	       += -lrt
+#
+# the order of these objects have to be swapped based on the GCC version
+MS_MPITMSG_O	= $(OUTDIR)/mpitmsg.o \
+		  $(OUTDIR)/ms.o 
+ifeq ($(GCC_MAJOR_VERSION_EQ_4),1)
+ifeq ($(GCC_MINOR_VERSION_GT_4),1)
+MS_MPITMSG_O	= $(OUTDIR)/ms.o \
```

Comment:
If any release above 4.4 should use this order of .sos this change won't work either. You may consider change it as follows
By default, the order should be ms.o and mpitmsg.o, 
Change the order when it is less than 4.4.